include/vigra/linear_solve.hxx

```diff
@@ -93,6 +95,41 @@ T determinantByLUDecomposition(MultiArrayView<2, T, C1> const & a)
     return det;
 }
 
+template <class T, class C1>
+typename NumericTraits<T>::Promote
+determinantByMinors(MultiArrayView<2, T, C1> const & mat)
+{
+    typedef typename NumericTraits<T>::Promote PromoteType;
+    MultiArrayIndex m = rowCount(mat);
+    MultiArrayIndex n = columnCount(mat);
+    vigra_precondition(
```

Comment:
I thought NumericTraits<T>::Promote is the promote type, where the inverse to the addition exists. So Promote(unsigned int) -> signed int.